j2me/core/src/org/javarosa/j2me/storage/rms/RMSStorageUtility.java

```diff
@@ -248,7 +246,7 @@ public void write (Persistable p) throws StorageFullException {
             if (!setReserveBuffer(bytesNeededEstimate)) {
                 setClean();
                 //can't reserve needed space to update index
-                throw new StorageFullException();
+                throw new RuntimeException();
```

Comment:
I think we may catch this upstream to do something specific, would be good to double check?